include/mbgl/util/platform.hpp

```diff
@@ -28,5 +29,12 @@ void showDebugImage(std::string name, const char *data, size_t width, size_t hei
 
 // Shows an alpha image with the specified dimensions in a named window.
 void showColorDebugImage(std::string name, const char *data, size_t logical_width, size_t logical_height, size_t width, size_t height);
+
+// Called when a thread is created, returns a specific thread data to be used when detaching the thread
+ThreadData attachThread();
```

Comment:
Can we move this to `include/mbgl/platform/thread.hpp`. The idea is to move all the platform implementation code to that folder.